cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/data_explorer/shared_links.spec.js

```diff
@@ -52,51 +52,58 @@ describe('shared links', () => {
     cy.waitForSearch();
   });
 
-  describe('shared links with state in query', () => {
-    it('should allow for copying the snapshot URL', function () {
-      const url = `http://localhost:5601/app/data-explorer/discover#/?_a=(discover:(columns:!(_source),isDirty:!f,sort:!()),metadata:(indexPattern:'logstash-*',view:discover))&_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:'2015-09-19T13:31:44.000Z',to:'2015-09-24T01:31:44.000Z'))&_q=(filters:!(),query:(language:kuery,query:''))`;
-      cy.getElementByTestId('shareTopNavButton').should('be.visible').click();
-      cy.getElementByTestId('copyShareUrlButton')
-        .invoke('attr', 'data-share-url')
-        .should('eq', url)
-        .then((url) => {
-          cy.log(url);
-          cy.request(url).its('status').should('eq', 200);
-        });
-    });
-
-    it('should allow for copying the snapshot URL as a short URL', function () {
-      cy.getElementByTestId('useShortUrl')
-        .should('be.visible')
-        .invoke('attr', 'aria-checked', 'true');
-      cy.wait(1000);
-      // For some reasons, after making the toggle true, the data-share-url do not get updated in the cypress test
-      // Thus only testing if the button is clickable, instead of comparing the actual URL
-      cy.getElementByTestId('copyShareUrlButton').should('be.visible').click();
-    });
-
-    it('should allow for copying the saved object URL', function () {
-      const url =
-        'http://localhost:5601/app/data-explorer/discover/#/view/ab12e3c0-f231-11e6-9486-733b1ac9221a?_g=%28filters%3A%21%28%29%2CrefreshInterval%3A%28pause%3A%21t%2Cvalue%3A0%29%2Ctime%3A%28from%3A%272015-09-19T13%3A31%3A44.000Z%27%2Cto%3A%272015-09-24T01%3A31%3A44.000Z%27%29%29';
-
-      cy.getElementByTestId('exportAsSavedObject')
-        .get('.euiRadio__input')
-        .should('be.disabled');
+  if (!Cypress.env('SECURITY_ENABLED')) {
```

Comment:
I am not a fan of this.

We could just strip the tenant from the query string and then compare. In fact, checking if the URL is identical is not a good idea; we should check if the elements we want are there, in whatever order.